### DISCUSSION TITLE
Reduce app size

### DISCUSSION THREAD

**Discussion by y1111** [FossifyOrg/General-Discussion#76] [Ideas]
first we could remove this locking/necessity to have the "thank you " app to use the shared themes . we could delete this app completely and remove the code about it in all apps.
so by default any app could use shared themes, with nothing more needed . 

and also it could have a separeted theme editor app just to edit the theme. and also compile a version in every app without the theme editor stuff inside the app(i think this could save 3-5 mb per app )  and using the sharded theme by default. this can be efficient to save redundancy of having this at any app . imagine having 5-10 fossify apps in the phone , u can edit the theme in a separeta app or install only 1 app with editor included and all the others without it .

**Comment by Aga-C** [comment]
See: https://github.com/FossifyOrg/General-Discussion/issues/45.

**Reply by y1111** [reply]
got it . u'll maintain the thanku so . if it will still exist and one need this app to use shared theme in any other app ,it makes sense to put the editor only in this app and remove for all the others  to streamline them. 

u get the idea , if a person cannot use the editor in app without thanku so why the editor still sits there.

**Reply by inson1** [reply]
he can use it without thank you app

**Reply by y1111** [reply]
also i do consider that some people maybe using only 1 app from the project so it makes sense having the editor . but maybe it could have 2 apks for each app like
app.apk
appst.apk 
//st - means using shared theme only. streamlined version with stripped editor , remove even theme option in the menu , just hardcode them to use shared theme.
 
like many projects that compile separated app32b.apk and app46bit.apk and post both in the releases . such small apps barely take 1-2min to compile will not be much hassle

this can even be automated like a "patch " code that for every realese also compiled the st one with those stripped

**Reply by inson1** [reply]
10MB per app is nothing, its not worth creating 2 apks to save 5MB (or even less) per app - for people and for developers, today people are talking in GB, nobody cares about MB, the lowest new phone should go is 128GB imo

so if you dont install it into fridge idk why do you so care about size of 2 photos - just delete 40 photos (or 20-10 if you have better camera)

**Comment by y1111** [comment]
as an ex. look at the dialer app. its ~9MB and its a so tiny and siomple app , theres dialers less than a MB
 https://github.com/SimpleMobileTools/Simple-Dialer/releases/tag/5.18.1 

keyboard app 18MB . 
those apps can certainly be reduced to half the current size

**Comment by naveensingh** [comment]
@y1111 

"Thank You" will be a paid app on the Google Play Store simply because there is no other way to accept donations there. It'll be available on F-Droid for free.

Assuming you meant 3-5 megabytes, why do you think a separate theme editor app will reduce individual app sizes by that much?

**Reply by y1111** [reply]
i think the apps are to big for what they are  . many have 3 screens and take at least 10-15MB . (like dialer , keyboard )
being the screens : 
-the main app scren, 
- settings tab 
- customize theme tab 

so removing one third scrennd may remove 1/3 size

**Reply by inson1** [reply]
@naveensingh hey could you point me to method in commons for theme icon editor? it would be great :) thank you very much

**Reply by y1111** [reply]
(too big comparing to others . like emerald launcher and dilaer , .SMSDroid , hacker keyboard . simplekeyboard-yeah this app have this same name but only812KB, etc  )
smsdroid ~7MB apk - 8 installed . simple sms ~9MB apk ~22 installed (android 10 armv8 64bit)

**Reply by inson1** [reply]
@y1111 keyboard - simple keyboard doesnt have emojis

**Reply by Aga-C** [reply]
@inson1 Simple Keyboard has emojis...

**Reply by inson1** [reply]
@Aga-C there are 2 simple keyboards - smt keyboard and simple keyboard, 1. does 2. doesnt

**Reply by y1111** [reply]
yeah , resuming , thats what i mean thats 2 with the same name look
https://search.f-droid.org/?q=simple+keyboard&lang=en
the smt 18MB - with emojis and theme 
the other 800kb no em no th

**Reply by inson1** [reply]
emojis take too much, Google keyboard uses about 100 MB

**Reply by inson1** [reply]
Most people want x and y feature and not 1MB decrease in size tbh
9MB or 18MB I dont see any difference at least in bigger picture, that is nothing, If it isnt at least 100MB I dont even think about it

**Comment by naveensingh** [comment]
Yes, we'll try to reduce the app size but it's not really a priority. Commons alone adds 2-5MB I think.
